### PR TITLE
[DR-1602] Update to use Java 11

### PR DESCRIPTION
**Pull request by RoriCremer**
The Data Repo with Java 11 (using the changes in this branch) is currently deployed to: https://jade-se.datarepo-dev.broadinstitute.org

These are the steps I took to to update my local environment:
1) Install Java 11: `brew install --cask adoptopenjdk11`
2) Update bash profile with `export JAVA_HOME=/Library/Java/JavaVirtualMachines/adoptopenjdk-11.jdk/Contents/Home`
3) If using IntelliJ, go to file > project structure and select Java 11 as the project sdk 
4) Run `./gradlew clean && ./gradlew assemble`

### CODE REVIEW

**Comment by samanehsan** [comment]
Ah it looks like the action failed because the branch name is `broadinstitute/datarepo-actions@rc-se-actions-testing`

File: build.gradle
Comment (samanehsan) [inline build.gradle]: The config options here mention `java8` is preferred for jdk 1.8+ (there is no `java11` option at the moment): https://openapi-generator.tech/docs/generators/java/

Comment (pshapiro4broad) [reply build.gradle]: This makes sense because the library was new in JDK 8 but hasn't been updated since. So java8 is really java8+

File: .github/workflows/int-and-connected-test-run.yml
Comment (nmalfroy) [inline .github/workflows/int-and-connected-test-run.yml]: Should we wait until the action is merged?

Comment (samanehsan) [reply .github/workflows/int-and-connected-test-run.yml]: Yes! I'll change this before merging in the PR (whenever we agree to do the upgrade)

**Comment by samanehsan** [comment]
I realized I still need to add the datarepo client tests as well since it uses a separate `build.gradle` file.

File: .github/workflows/int-and-connected-test-run.yml
Comment (tlangs) [inline .github/workflows/int-and-connected-test-run.yml]: should this be using what looks like a personal branch?

File: datarepo-clienttests/build.gradle
Comment (tlangs) [inline datarepo-clienttests/build.gradle]: Any reason why we don't want to use the latest version `4.7.0`?

File: datarepo-clienttests/build.gradle
Comment (tlangs) [inline datarepo-clienttests/build.gradle]: Do we want to use the latest version of `4.2.2`?

**Review by tlangs — CHANGES_REQUESTED** [review]

File: .github/workflows/int-and-connected-test-run.yml
Comment (tlangs) [reply .github/workflows/int-and-connected-test-run.yml:51]: Doh, Muscles already caught this, ignore me.

Comment (fboulnois) [reply .github/workflows/int-and-connected-test-run.yml]: (my guess is @samanehsan / @RoriCremer will switch back when ready)

File: docs/jade-getting-started.md
Comment (fboulnois) [inline docs/jade-getting-started.md]: I think you might need to rebase to get the latest changes to the getting started doc.

**Review by fboulnois — APPROVED** [review]
Do we need to make any changes to our Java code? Or does it just work out of the box?

**Comment by samanehsan** [comment]
> Do we need to make any changes to our Java code? Or does it just work out of the box?

@fboulnois, surprisingly I didn't run into any errors when running the server, but I am seeing this warning: 
`WARNING: Illegal reflective access by org.springframework.cglib.core.ReflectUtils (file:/Users/sehsan/.gradle/caches/modules-2/files-2.1/org.springframework/spring-core/5.2.9.RELEASE/400a6fdb45bfa5318aa7d06360f4495b75080bb5/spring-core-5.2.9.RELEASE.jar) to method java.lang.ClassLoader.defineClass(java.lang.String,byte[],int,int,java.security.ProtectionDomain)`

**Review by tlangs — APPROVED** [review]

**Pull request merged**
